### PR TITLE
Remove duplicated plausible props

### LINKED ISSUE #2534: Default plausible prop payloads overlap with built-in Plausible stats data
## Description

Plausible provides a number of useful built-in event data that can be used for querying the stats API: https://plausible.io/docs/stats-api#properties

While these do not show visually in the graphical custom event view on the Plausible frontend, they have the distinct advantage over custom event payload props that they can be combined. Whereas custom event payload props can only be filtered against one at a time, the built in event data can be combined to filter against multiple data points.

Additionally, having overlap not only unnecessarily complicates our code (and duplicates built in features), it also makes it confusing for folks querying our events to know whether to use the Plausible built in event data or the payload props.

## Fix

Remove the following overlapping default payload props from the default props added in the `useAnalytics` composable. I've noted which built in event data they overlap with.

- `browser` built in as `visit:browser`
- `version` built in as `visit:browser_version`
- `platform` built in as `visit:os`
- `path` built in as `event:page`
- `origin` built in as `event:referrer`
  - Note: For pageviews this would be where the user came from (i.e., if they followed a link from Twitter to Openverse, it would be twitter.com). For custom events, however, it's just the origin of the page making the request and therefore matches the behaviour of the `origin` custom payload prop.
- `timestamp`, Plausible already records the timestamp of all events and has special tools for querying events as timeseries data. The `events` table Plausible creates includes a timestamp _column_ which is far more useful for querying against than custom prop data.

All of these can be confirmed by checking the `events` table in the local Clickhouse database for our local Plausible instances:

```
DESCRIBE TABLE events

┌─name─────────────────────┬─type───────────────────────────┬─default_type─┬─default_expression─┬─comment─┬─codec_expression─┬─ttl_expression─┐
│ name                     │ String                         │              │                    │         │                  │                │
│ domain                   │ String                         │              │                    │         │                  │                │
│ user_id                  │ UInt64                         │              │                    │         │                  │                │
│ session_id               │ UInt64                         │              │                    │         │                  │                │
│ hostname                 │ String                         │              │                    │         │                  │                │
│ pathname                 │ String                         │              │                    │         │                  │                │
│ referrer                 │ String                         │              │                    │         │                  │                │
│ referrer_source          │ String                         │              │                    │         │                  │                │
│ country_code             │ LowCardinality(FixedString(2)) │              │                    │         │                  │                │
│ screen_size              │ LowCardinality(String)         │              │                    │         │                  │                │
│ operating_system         │ LowCardinality(String)         │              │                    │         │                  │                │
│ browser                  │ LowCardinality(String)         │              │                    │         │                  │                │
│ timestamp                │ DateTime                       │              │                    │         │                  │                │
│ utm_medium               │ String                         │              │                    │         │                  │                │
│ utm_source               │ String                         │              │                    │         │                  │                │
│ utm_campaign             │ String                         │              │                    │         │                  │                │
│ meta.key                 │ Array(String)                  │              │                    │         │                  │                │
│ meta.value               │ Array(String)                  │              │                    │         │                  │                │
│ browser_version          │ LowCardinality(String)         │              │                    │         │                  │                │
│ operating_system_version │ LowCardinality(String)         │              │                    │         │                  │                │
│ subdivision1_code        │ LowCardinality(String)         │              │                    │         │                  │                │
│ subdivision2_code        │ LowCardinality(String)         │              │                    │         │                  │                │
│ city_geoname_id          │ UInt32                         │              │                    │         │                  │                │
│ utm_content              │ String                         │              │                    │         │                  │                │
│ utm_term                 │ String                         │              │                    │         │                  │                │
│ transferred_from         │ String                         │              │                    │         │                  │                │
└──────────────────────────┴────────────────────────────────┴──────────────┴────────────────────┴─────────┴──────────────────┴────────────────┘
```

Example of `LOAD_MORE_RESULTS` event locally:

```
SELECT *
FROM events
WHERE name = 'LOAD_MORE_RESULTS'

┌─name──────────────┬─domain────┬─────────────user_id─┬─────────session_id─┬─hostname──────┬─pathname─┬─referrer─┬─referrer_source─┬─country_code─┬─screen_size─┬─operating_system─┬─browser─┬───────────timestamp─┬─utm_medium─┬─utm_source─┬─utm_campaign─┬─meta.key──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┬─meta.value─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┬─browser_version─┬─operating_system_version─┬─subdivision1_code─┬─subdivision2_code─┬─city_geoname_id─┬─utm_content─┬─utm_term─┬─transferred_from─┐
│ LOAD_MORE_RESULTS │ localhost │ 4418127662590562235 │ 717148294291195879 │ 192.168.0.226 │ /search/ │          │                 │              │ Laptop      │ GNU/Linux        │ Firefox │ 2023-07-02 23:33:49 │            │            │              │ ['breakpoint','browser','height','language','origin','os','pathname','platform','query','resultPage','searchType','timestamp','ua','version','width'] │ ['xl','Firefox','951','en','http://192.168.0.226:8443','Linux 64','/search/','Linux','galah','1','all','2023-07-02T23:33:47.783Z','Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/114.0','114.0','1365'] │ 114.0           │                          │                   │                   │               0 │             │          │                  │
│ LOAD_MORE_RESULTS │ localhost │ 4418127662590562235 │ 717148294291195879 │ 192.168.0.226 │ /search/ │          │                 │              │ Laptop      │ GNU/Linux        │ Firefox │ 2023-07-02 23:33:55 │            │            │              │ ['breakpoint','browser','height','language','origin','os','pathname','platform','query','resultPage','searchType','timestamp','ua','version','width'] │ ['xl','Firefox','951','en','http://192.168.0.226:8443','Linux 64','/search/','Linux','galah','3','all','2023-07-02T23:33:47.783Z','Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/114.0','114.0','1365'] │ 114.0           │                          │                   │                   │               0 │             │          │                  │
│ LOAD_MORE_RESULTS │ localhost │ 4418127662590562235 │ 717148294291195879 │ 192.168.0.226 │ /search/ │          │                 │              │ Laptop      │ GNU/Linux        │ Firefox │ 2023-07-02 23:33:52 │            │            │              │ ['breakpoint','browser','height','language','origin','os','pathname','platform','query','resultPage','searchType','timestamp','ua','version','width'] │ ['xl','Firefox','951','en','http://192.168.0.226:8443','Linux 64','/search/','Linux','galah','2','all','2023-07-02T23:33:47.783Z','Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/114.0','114.0','1365'] │ 114.0           │                          │                   │                   │               0 │             │          │                  │
└───────────────────┴───────────┴─────────────────────┴────────────────────┴───────────────┴──────────┴──────────┴─────────────────┴──────────────┴─────────────┴──────────────────┴─────────┴─────────────────────┴────────────┴────────────┴──────────────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴─────────────────┴──────────────────────────┴───────────────────┴───────────────────┴─────────────────┴─────────────┴──────────┴──────────────────┘
```

As you can see, much of the data in the `meta.value` column (custom event payloads) duplicates data that have their own proper columns (which, again, are all much more flexible when querying than custom payload data).

To implement this, all that is needed is to delete the lines in the `useAnalytics` composable that add them to the payload props.

I believe we could safely remove the `ua` prop as well as the relevant data from it is broken out into the built in event information. What do you think @dhruvkb, should `ua` be dropped as well? We wouldn't really be able to query against it more effectively than the broken out browser and platform data.

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #2534 by @sarayourfriend 

## Description

Removes custom props added as defaults to all plausible events in favor of corresponding default values already provided by plausible.

## Testing Instructions

Read the issue and make sure the removed events are correct. All tests should pass.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/composables/use-analytics.ts [hunk 1]
```diff
@@ -73,7 +60,7 @@ export const useAnalytics = () => {
       props: {
         ...isomorphicProps.value,
         ...windowProps.value,
-        ...payload, // can override mandatory props
+        ...payload, // will override matching mandatory props
```
Comment (sarayourfriend) [inline frontend/src/composables/use-analytics.ts:63]: It's actually not true now that the ESLint rule disallows it :slightly_smiling_face: 

Speaking of which, you'll need to remove these prop names from the ESLint rule. Or should we keep them/change the rule to infer from the name that we're accidentally duplicating built-in event data?

**Review by sarayourfriend — CHANGES_REQUESTED** [review]
LGTM, just one note about the ESLint rule.

File: frontend/src/composables/use-analytics.ts (hunk 1 above)
Comment (zackkrida) [reply frontend/src/composables/use-analytics.ts]: I've removed the comment and the prop names from the Eslint rule for now, thanks!

The separate issue sounds like a welcome addition, a "you might be duplicating a built-in prop" notification. I wonder if name inference would be reliable enough to be useful? Like if someone tried to add a prop called `userAgent`, for example, that wouldn't match `ua`.

Comment (sarayourfriend) [reply frontend/src/composables/use-analytics.ts]: Issue here: https://github.com/WordPress/openverse/issues/2597

**Review by sarayourfriend — APPROVED** [review]
LGTM!

I realised when re-reviewing that `language` could be derived from the built-in `page`, as we can derive the locale from locale prefix (or `en` if there isn't one). It's probably easier to query along those lines too because you can't combine custom prop filters to see, for example, if a "load more" events on image queries happen less for users of the RTL version of the site than LTR, because you couldn't query against both `mediaType` _and_ `language`. However, you _can_ filter `page` using wildcards, so you could filter, for example, `event:page=/ar*&event:name=LOAD_MORE_RESULTS&event:props:mediaType=image` to grab the event when it happened on the Arabic version of the site.

Anyway, not a blocker, but if you think it's worth removing and adding documentation to this effect somewhere, another issue would be good.

**Review by dhruvkb — APPROVED** [review]
It would be helpful to have some comment pointing to the Plausible docs listing the default params they provide.

LGTM otherwise.

**Comment by sarayourfriend** [comment]
> I can't find anything relating to this from Plausible! I'll ask them and create a new issue.

All of this is documented here, isn't it? https://plausible.io/docs/stats-api#properties (Linked in the issue, by the way)

Timestamp is the only think missing, as far as I can tell, but you can't query against timestamps directly from Plausible's API.

**Comment by zackkrida** [comment]
That's perfect, thanks!

**Pull request merged**